### DISCUSSION TITLE
Create repository via GraphQL

### DISCUSSION THREAD

**Discussion by CraigMilhiser** [community/community#24395] [Apps, API and Webhooks]
<p>From here - <a href="https://platform.github.community/t/schema-request-create-repository-and-get-repository-contents/1833" rel="nofollow noopener">https://platform.github.community/t/schema-request-create-repository-and-get-repository-contents/1833</a>, creating a repo via GraphQL was not yet supported. The last update on that thread was Feb 8, 2018, and since the community has moved platforms.</p>
<p>Is there any update if create repository is now supported in GraphQL? I do not see a createRepository mutation in the documentation but perhaps there is another method.</p>
<p>Thanks</p>

**Comment by ThirdEyeRose** [comment, marked as answer]
<p>Hi <a class="mention" href="/craigmilhiser">@craigmilhiser</a>,</p>
<p>Thanks for asking! You are correct, there is not currently a method for creating repositories using the GraphQL API. To create a repository, you will still need to use the <a href="https://developer.github.com/v3/repos/#create" rel="nofollow noopener">Create a repository v3 API endpoint</a>.</p>
<p>Thanks!</p>

**Comment by NathanZook** [comment]
<p>I would suggest placing this information directly on the v4 documentation.  It would save a lot of time.</p>
<p>More generally, if there were some statement regarding the up-to-dated-ness of the documented schema, that would be great.  In particular, I would hope that the documentation would be autogenereated whenever a change goes live, but that might be easier said than done.</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/nathanzook">@nathanzook</a>,</p>
<p>As we’re always trying to make GitHub friendlier and more approachable, I appreciate that recommendation. I will pass that on to our documentation team for consideration.</p>
<p>Thanks!</p>

**Comment by jsjoeio** [comment]
<p><a class="mention" href="/that-pat">@that-pat</a> when will it be possible to create a repository via the v4 API?</p>

**Comment by AndreaGriffiths11** [comment]
<p>You can now create a new repository using the following mutation: <a href="https://developer.github.com/v4/mutation/createrepository/" rel="nofollow noopener">https://developer.github.com/v4/mutation/createrepository/</a></p>

**Reply by jef** [reply]
Thanks!
